test/unit/core/type_traits/range_iterator_test.cpp

```diff
@@ -142,24 +142,18 @@ TEST(range_and_iterator, rvalue_reference_)
 
 TEST(range_and_iterator, const_reference_)
 {
-//     using iterator_of_int_vector = std::ranges::iterator_t<std::vector<int>>;
-//     using foreign_iterator = seqan3::detail::random_access_iterator<std::vector<int>>;
+    using const_iterator_of_int_vector = std::ranges::iterator_t<std::vector<int> const>;
+    using const_foreign_iterator = seqan3::detail::random_access_iterator<std::vector<int> const>;
     auto v = std::views::iota(1);
-    using type_list_example = seqan3::type_list<seqan3::const_reference_t<std::vector<int>>, // short
-                                                typename seqan3::const_reference<std::vector<int>>::type, // long
+    using const_iterator_of_view = std::ranges::iterator_t<decltype(v) const>;
+    using type_list_example = seqan3::type_list<std::iter_reference_t<const_iterator_of_int_vector>, // short
```

Comment:
This one was about a container, so we should use `std::ranges::range...`:
```suggestion
    using type_list_example = seqan3::type_list<std::ranges::range_reference_t<std::vector<int> const>, // short
```

`std::iter_reference_t<const_iterator_of_int_vector>` should be grouped with the ones that come later